src/components/global/Navbar.jsx

```diff
@@ -1,8 +1,147 @@
-const Navbar = () => {
-  // any logic needed
-  // return (
-  // all the JSX (looks like HTML) to render
-  // )
+import {
+  AppBar,
+  Box,
+  Hidden,
+  IconButton,
+  List,
+  ListItem,
+  Popover,
+  Toolbar,
+  Typography
+} from '@mui/material';
+import MenuIcon from '@mui/icons-material/Menu';
+import { Brightness4, Brightness7 } from '@mui/icons-material';
+import { BrowserRouter as Route, Link } from 'react-router-dom';
+import { useState, useRef } from 'react';
+
+const logoBlobStyle = {
+  position: 'absolute',
+  bottom: 0,
+  left: -30,
+  right: 0,
+  top: -50,
+  height: '250px',
+  backgroundImage: { md: 'url(/assets/logoBlobNav.svg)', sm: 'none' },
+  backgroundRepeat: 'no-repeat',
+  display: {
+    lg: 'flex',
+    xs: 'none'
+  },
+  maxWidth: '200px'
+};
+
+const textStyle = {
+  textDecoration: 'none',
+  color: 'inherit'
 };
 
-export default Navbar;
+function NavBar() {
+  const [darkMode, setDarkMode] = useState(false);
+  const [menuOpen, setMenuOpen] = useState(false);
+
+  // anchor for menu to pop from its icon
+  const anchorRef = useRef(null);
+
+  const handleThemeChange = () => {
+    setDarkMode(!darkMode);
+  };
+
+  const handleToggle = () => {
+    setMenuOpen((prevOpen) => !prevOpen);
+  };
+
+  const handleClose = () => {
+    setMenuOpen(false);
+  };
+
+  const menuItems = (
+    <List>
+      <ListItem component={Link} to="/" onClick={handleClose}>
+        Home
+      </ListItem>
+      <ListItem component={Link} to="/projects" onClick={handleClose}>
+        Projects
+      </ListItem>
+      <ListItem component={Link} to="/volunteer" onClick={handleClose}>
+        Volunteer
+      </ListItem>
+    </List>
+  );
+
+  return (
+    <AppBar position="static" color={darkMode ? 'dark' : 'primary'}>
+      <Toolbar sx={{ height: '100px', alignItems: 'center' }}>
+        <Box sx={logoBlobStyle}></Box>
+        <Link to="/" style={textStyle} aria-label="Home">
+          <Box
+            component="img"
```

Comment:
Do you plan on abstracting more styling as we have elsewhere? No major issue, just wondering.

Reply:
a fair chunk of it is just unique enough where it wont save any code, really improve readability so I left it. some of it has to be done at the component level.